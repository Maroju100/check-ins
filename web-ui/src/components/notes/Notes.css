.notes {
}

.container {
  display: flex;
  justify-content: center;
}

.container textarea {
  border: 2px solid black;
  min-height: 150px;
  padding: 5px;
  width: 50%;
}
