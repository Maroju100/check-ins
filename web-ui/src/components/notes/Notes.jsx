import React, { useContext, useEffect, useState } from "react";
import { AppContext } from "../../context/AppContext";
import {getCheckinByMemberId} from '../../api/checkins'

const Notes = () => {
  const { state } = useContext(AppContext);
  const { userProfile } = state;
  const { id } = userProfile;
  const [notes, setNotes] = useState([])
  const [checkin, setCheckin] = useState({})
  
  useEffect(() => {
    async function updateCheckin() {
      if (id) {
        let res = await getCheckinByMemberId(id);
        let data =
          res.payload &&
          res.payload.data &&
          res.payload.status === 200 &&
          !res.error
            ? res.payload.data
            : null;
        if (data) {
          setCheckin(data);
        }
      }
    }
    updateCheckin();
  }, [id]);

  useEffect(() => {
    async function updateNotes() {
      if (checkin.id) {
        let res = await getCheckinByMemberId(id);
        let data =
          res.payload &&
          res.payload.data &&
          res.payload.status === 200 &&
          !res.error
            ? res.payload.data
            : null;
        if (data) {
          setCheckin(data);
        }
      }
    }
    updateNotes();
  }, [checkin]);

  return <div>Notes</div>;
};

export default Notes;
