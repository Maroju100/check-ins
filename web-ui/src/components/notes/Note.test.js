import React from "react";
import Notes from "./Note";
// import { render, fireEvent, screen } from "@testing-library/react";

let checkin = {
  id: "3a1906df-d45c-4ff5-a6f8-7dacba97ff1a",
  checkinid: "bf9975f8-a5b2-4551-b729-afd56b49e2cc",
  createdbyid: "5425d835-dcd1-4d91-9540-200c06f18f28",
  description: "updated string",
};

it("renders correctly with checkin", () => {
  snapshot(<Notes checkin={checkin} />);
});
