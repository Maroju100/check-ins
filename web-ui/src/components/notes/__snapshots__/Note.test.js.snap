// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with checkin 1`] = `
<div
  className="notes"
>
  <div>
    <h1>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "marginRight": "10px",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h12v-2H3v2zM3 6v2h18V6H3zm0 7h18v-2H3v2z"
        />
      </svg>
      Notes for 
    </h1>
    <div
      className="container"
    >
      <textarea
        onChange={[Function]}
      />
    </div>
  </div>
  <div>
    <h1>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "marginRight": "10px",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
        />
      </svg>
      Private Notes
    </h1>
    <div
      className="container"
    >
      <textarea
        onChange={[Function]}
        value="Private note"
      >
        <p />
      </textarea>
    </div>
  </div>
</div>
`;
